src/classes/CaseTriggerHandler.cls

```diff
@@ -0,0 +1,135 @@
+/**
+ * Created by Synebo on 14.09.2021.
+ */
+
+public with sharing class CaseTriggerHandler extends TriggerHandler {
+
+    public override void afterUpdate() {
+        this.countCaseUpdate((List<Case>) Trigger.new, (Map<Id, Case>) Trigger.oldMap);
+    }
+
+    public override void afterInsert() {
+        this.countCaseInsert((List<Case>) Trigger.new);
+    }
+
+    public override void afterUndelete() {
+        this.countCaseInsert((List<Case>) Trigger.new);
+    }
+    public override void afterDelete() {
+        this.countCaseDelete((Map<Id, Case>) Trigger.oldMap);
+    }
+
+    private void countCaseDelete(Map<Id, Case>oldMap){
+        Map<Id, String> caseWithStatus = new Map<Id, String>();
+        if (!oldMap.isEmpty()) {
+            for (Case case_i : oldMap.values()) {
+                if (case_i.Status != null && case_i.ContactId != null) {
```

Comment:
Here you use a Status, but in countCaseUpdate you use IsClosed. What way is right?